packages/flutter_tools/lib/src/xcode_project.dart

```diff
@@ -134,6 +134,12 @@ class IosProject extends XcodeBasedProject {
 
   Directory get symlinks => _flutterLibRoot.childDirectory('.symlinks');
 
+  /// True, if the app project is using swift.
+  bool get isSwift {
```

Comment:
cc @jmagman

Reply:
We have this, but it needs to be run on a Mac with Xcode installed.
https://github.com/flutter/flutter/blob/320d2cff6433415315dad1efe791e727f7bbd6c0/packages/flutter_tools/lib/src/macos/cocoapods.dart#L246-L249

I'm guessing this needs to work on all platforms, like if you ran this on Windows you would still want the `ios` directory migrated.

A user could rename `AppDelegate.swift` but they likely wouldn't.

Reply:
Yes. All platform compatibility is important. I can try to include a few more fallback checks if that is something that could help with robustness.